Big data bug fixes

BlankCanvasStudio opened the pull request:
The larger optimization problems using the conic solver were being rejected due to a negative dimension. This arose from the int32 datatype being used. I changed a couple int32 data types to int64 data types so the overflows wouldn't occur. This completely fixed the problems and the solvers seem to be running without any other changes. This fix can definitely be implemented individually in other solvers but I will look through the code in the future to try and cast all int32 types to int64 types to avoid this problem.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cvxgrp/cvxpy?pullRequest=1074) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cvxgrp/cvxpy?pullRequest=1074) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cvxgrp/cvxpy?pullRequest=1074) it.</sub>

BlankCanvasStudio commented:
So this build failed due to the following error:

Collecting package metadata (current_repodata.json): failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://conda.anaconda.org/conda-forge/linux-64/current_repodata.json>

Elapsed: -

An HTTP error occurred when trying to retrieve this URL.

HTTP errors are often intermittent, and a simple retry will get you on your way.

'https://conda.anaconda.org/conda-forge/linux-64'

This doesn't have anything to do with the errors I fixed. my changes were only in 2 files that needed to be downloaded, nothing to do with the download protocol. I don't know why these errors are popping up but it doesn't seem to be cause by the two lines I changed for this fix. If someone could please look this over and let me know why it failed I'd be more than willing to fix it. Thanks!

SteveDiamond commented:
I restarted the test that failed. This happens sometimes, where it can't download a package etc. Of course not due to your change.

SteveDiamond commented:
Let's merge this in before we make further changes. Can you sign the CLA?

BlankCanvasStudio commented:
I've tried to sign the CLA but it keeps me stuck in a loop on the agreement
page (for lack of a better phrase). I click the 'sign in with github to
agree' button and it simply reloads that page again. I don't know why I'm
stuck in this loop or how to fix it.

Thanks,
Spencer Stingley

On Thu, Jul 9, 2020 at 12:48 PM Steven Diamond <notifications@github.com>
wrote:

> Let's merge this in before we make further changes. Can you sign the CLA?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cvxgrp/cvxpy/pull/1074#issuecomment-656317579>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMCXNWGJKNW6XY6EYAOBF3R2YNHLANCNFSM4OV4XSYA>
> .
>


-- 
Thanks,
Spencer Stingley

SteveDiamond commented:
Ok, it's a very small change so I'll just merge it.

BlankCanvasStudio commented:
Ok. I'll try to look through the rest of the code to fix the other int32
issues people have brought up and I'll send you an email if I see a fix to
the problem.

Thanks,
Spencer Stingley

On Thu, Jul 9, 2020, 2:53 PM Steven Diamond <notifications@github.com>
wrote:

> Merged #1074 <https://github.com/cvxgrp/cvxpy/pull/1074> into master.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cvxgrp/cvxpy/pull/1074#event-3531216732>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMCXNUVIZQCABY7HMNPA3TR2YN2NANCNFSM4OV4XSYA>
> .
>
